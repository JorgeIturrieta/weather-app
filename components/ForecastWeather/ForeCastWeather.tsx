import getHumanReadableTime from 'lib/getHumanReadableTime'
import React, { FunctionComponent } from 'react'
import { Daily } from '../../types/weatherApi/apiWeatherForecast.interface'
import style from './ForecastWeather.module.scss'
type ForeCastWeatherProps = {
  data: Daily[]
  timezone: number
}

const ForeCastWeather: FunctionComponent<ForeCastWeatherProps> = ({ data, timezone }) => {
  return (
    <div>
      <h2>Pronóstico extendido</h2>
      <div className={style.containerGrid}>
        {data.map((day, idx) => (
          <div key={idx} className={style.gridItem}>
            <h3>
              {getHumanReadableTime(day.dt, timezone, {
                day: 'numeric',
                weekday: 'long',
                month: 'long',
              })}
            </h3>
            <p>Máx: {day.temp.max}°</p>
            <p>Min: {day.temp.min}°</p>
            <p>
              Amanecer:{' '}
              {getHumanReadableTime(day.sunrise, timezone, {
                hour: 'numeric',
                minute: 'numeric',
              })}
              hs
            </p>
            <p>
              Atardecer:{' '}
              {getHumanReadableTime(day.sunset, timezone, {
                hour: 'numeric',
                minute: 'numeric',
              })}
              hs
            </p>
          </div>
        ))}
      </div>
    </div>
  )
}

export default ForeCastWeather
