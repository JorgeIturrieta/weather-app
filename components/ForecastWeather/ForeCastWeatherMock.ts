import { WeatherForecastAdapted } from '../../types/weatherForecast.interface'
export const weatherForecastMock: WeatherForecastAdapted[] = [
  {
    day: 'jueves, 9 de junio',
    temp_max: 18.68,
    temp_min: 9.8,
    sunrise: '8:29',
    sunset: '18:37',
  },
  {
    day: 'viernes, 10 de junio',
    temp_max: 13.66,
    temp_min: 7.24,
    sunrise: '8:29',
    sunset: '18:37',
  },
  {
    day: 's√°bado, 11 de junio',
    temp_max: 19.91,
    temp_min: 7.83,
    sunrise: '8:30',
    sunset: '18:37',
  },
  {
    day: 'domingo, 12 de junio',
    temp_max: 21.22,
    temp_min: 9.06,
    sunrise: '8:30',
    sunset: '18:37',
  },
  {
    day: 'lunes, 13 de junio',
    temp_max: 19.43,
    temp_min: 9.52,
    sunrise: '8:31',
    sunset: '18:37',
  },
]
