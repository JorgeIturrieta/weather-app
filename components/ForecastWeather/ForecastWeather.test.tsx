import { render, screen, within } from '@testing-library/react'
import ForeCastWeather from './ForeCastWeather'
import { weatherForecastMock } from './ForeCastWeatherMock'
describe('<ForecastWeather />', () => {
  it('Render ForecastWeather and should render only 5 items', async () => {
    render(<ForeCastWeather data={weatherForecastMock} />)

    const mainHeading = screen.getByRole('heading', { name: /pronÃ³stico extendido/i })
    const containerDay = screen.getByTestId('list')
    const { getAllByTestId } = within(containerDay)
    const items = getAllByTestId('listitem')
    expect(mainHeading).toBeInTheDocument()
    expect(items.length).toBe(5)
  })

  it('renders ForeCastWeather unchanged', () => {
    const { container } = render(<ForeCastWeather data={weatherForecastMock} />)
    expect(container).toMatchSnapshot()
  })
})
