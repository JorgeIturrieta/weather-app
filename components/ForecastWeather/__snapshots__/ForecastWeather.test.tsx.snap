// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForecastWeather /> renders ForeCastWeather unchanged 1`] = `
<div>
  <div>
    <h2>
      Pronóstico extendido
    </h2>
    <div
      class="containerGrid"
      data-testid="list"
    >
      <div
        class="gridItem"
        data-testid="listitem"
      >
        <h3>
          jueves, 9 de junio
        </h3>
        <p>
          Máx: 
          18.68
          °
        </p>
        <p>
          Min: 
          9.8
          °
        </p>
        <p>
          Amanecer: 
          8:29
          hs
        </p>
        <p>
          Atardecer: 
          18:37
          hs
        </p>
      </div>
      <div
        class="gridItem"
        data-testid="listitem"
      >
        <h3>
          viernes, 10 de junio
        </h3>
        <p>
          Máx: 
          13.66
          °
        </p>
        <p>
          Min: 
          7.24
          °
        </p>
        <p>
          Amanecer: 
          8:29
          hs
        </p>
        <p>
          Atardecer: 
          18:37
          hs
        </p>
      </div>
      <div
        class="gridItem"
        data-testid="listitem"
      >
        <h3>
          sábado, 11 de junio
        </h3>
        <p>
          Máx: 
          19.91
          °
        </p>
        <p>
          Min: 
          7.83
          °
        </p>
        <p>
          Amanecer: 
          8:30
          hs
        </p>
        <p>
          Atardecer: 
          18:37
          hs
        </p>
      </div>
      <div
        class="gridItem"
        data-testid="listitem"
      >
        <h3>
          domingo, 12 de junio
        </h3>
        <p>
          Máx: 
          21.22
          °
        </p>
        <p>
          Min: 
          9.06
          °
        </p>
        <p>
          Amanecer: 
          8:30
          hs
        </p>
        <p>
          Atardecer: 
          18:37
          hs
        </p>
      </div>
      <div
        class="gridItem"
        data-testid="listitem"
      >
        <h3>
          lunes, 13 de junio
        </h3>
        <p>
          Máx: 
          19.43
          °
        </p>
        <p>
          Min: 
          9.52
          °
        </p>
        <p>
          Amanecer: 
          8:31
          hs
        </p>
        <p>
          Atardecer: 
          18:37
          hs
        </p>
      </div>
    </div>
  </div>
</div>
`;
